transformer_engine/jax/cpp_extensions.py

```diff
@@ -0,0 +1,1975 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+# pylint: disable=c-extension-no-member
+# pylint: disable=no-name-in-module
```

Comment:
I saw pylint reports that warning but didn't know how to resolve it even I google that warnings. Thus, I suppress them. Do you have expereinces on `c-extension-no-member` and `no-name-in-module`? Both them are regarding Python CPP binding.

Reply:
I'm just suspicious of disabling warnings if we don't understand what's causing them. It's just asking for problems to show up in the future.

Did you get any more info about where it's happening? It makes me think something isn't being imported correctly. I wonder if it's because pylint is running in an environment without JAX?

Reply:
I saw a workaround here: https://stackoverflow.com/questions/28437071/pylint-1-4-reports-e1101no-member-on-all-c-extensions.

In short reason:

> Pylint was decided that importing c extensions that are not part of the python stdlib is a security risk and could introduce malicious code.

The workaround:
`extension-pkg-whitelist=transformer_engine_jax` in `pylint.rc` to disable warning

Reply:
Hmm, it is strange that why warnings were not triggered by TE/PyTorch

Reply:
After trial and error for hours, I still don't know why warnings were not triggered by TE/PyTorch.
The pylint message also said:
```
transformer_engine/jax/cpp_extensions.py:24:21: I1101: Module 'transformer_engine_jax' has no 'registrations' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
```
Thus, I think adding extension-pkg-whitelist=transformer_engine_jax in pylint.rc is best approach for now.
Does it make sense?

Reply:
@timmoon10 ,
This is why TE/PyTorch didn't trigger the warning: [qa/L0_lint/pylintrc#L2](https://github.com/NVIDIA/TransformerEngine/blob/main/qa/L0_lint/pylintrc#L2)

```python
extension-pkg-whitelist=torch,
                        transformer_engine_extensions
```
I will add `transformer_engine_jax` to `extension-pkg-whitelist` and push a commit later

Reply:
This issue should be solved by [252b8bf](https://github.com/NVIDIA/TransformerEngine/pull/53/commits/252b8bfd953abc552f00d7d87a4e6a28abc68448)

Reply:
Nice debugging, this is a much cleaner solution.